src/Http/Controllers/GraphqlController.php

```diff
@@ -16,21 +16,32 @@
 
 class GraphqlController extends Controller {
 
-	protected Application $app;
+	protected IPackages $packages;
 
-	public function __construct(Application $app)
+	public function __construct(IPackages $packages)
 	{
-		$this->app = $app;
+		$this->packages = $packages;
 	}
 
+	/**
+	 * @return \Illuminate\Http\Response|\Illuminate\Http\JsonResponse|\Laravel\Lumen\Http\ResponseFactory|mixed
+	 */
 	public function run(Request $request)
 	{
-		$builtin = Parser::parse(file_get_contents(package()->path("buildin-definations.graphql")));
+		$builtinFileContent = file_get_contents(package()->path("buildin-definations.graphql"));
+		if ($builtinFileContent === false) {
+			throw new \RuntimeException("can not read content of file 'buildin-definations.graphql'");
+		}
+		$builtin = Parser::parse($builtinFileContent);
 		$schema = BuildSchema::build($builtin);
-		foreach ($this->app->packages->all() as $package) {
+		foreach ($this->packages->all() as $package) {
+			/** @var \Jalno\Lumen\Contracts\IPackage&IGraphQLable $package */
```

Comment:
phpstan can't understand it?

Reply:
No, it's thought the `$package` is JUST an instance of `IGraphQLable` and triggers an error that says: `call to undefined function path on $package`.\
I search for that and find the `instanceof` is the best and fastest operator for this type of operation and the solution is to use union types.